File: fortran/tests/test-auger.f90
PROGRAM test_auger

USE, INTRINSIC :: ISO_C_BINDING
USE, INTRINSIC :: ISO_FORTRAN_ENV
USE :: xraylib
USE :: libtest
IMPLICIT NONE

TYPE(xrl_error), POINTER :: error => NULL()
REAL (C_DOUBLE) :: rate, yield

rate = AugerRate(82, K_L3M5_AUGER, error)
CALL assert(ABS(rate - 0.004573193387_C_DOUBLE) < 1E-6_C_DOUBLE)
CALL assert(.NOT. ASSOCIATED(error))

rate = AugerRate(82, L3_M4N7_AUGER, error)
CALL assert(ABS(rate - 0.0024327572005_C_DOUBLE) < 1E-6_C_DOUBLE)
CALL assert(.NOT. ASSOCIATED(error))

rate = AugerRate(-35, L3_M4N7_AUGER, error)
CALL assert(rate == 0.0_C_DOUBLE)
CALL assert(ASSOCIATED(error))
CALL assert(error%code == XRL_ERROR_INVALID_ARGUMENT)
WRITE (output_unit, '(A,A)') 'Error message: ', TRIM(error%message)
DEALLOCATE(error)

rate = AugerRate(82, M4_M5Q3_AUGER + 1, error)
CALL assert(rate == 0.0_C_DOUBLE)
CALL assert(ASSOCIATED(error))
CALL assert(error%code == XRL_ERROR_INVALID_ARGUMENT)
WRITE (output_unit, '(A,A)') 'Error message: ', TRIM(error%message)
DEALLOCATE(error)

rate = AugerRate(62, L3_M4N7_AUGER, error)
CALL assert(rate == 0.0)
CALL assert(ASSOCIATED(error))
CALL assert(error%code == XRL_ERROR_INVALID_ARGUMENT)
WRITE (output_unit, '(A,A)') 'Error message: ', TRIM(error%message)
DEALLOCATE(error)

yield = AugerYield(82, K_SHELL, error)
CALL assert(.NOT. ASSOCIATED(error))
CALL assert(ABS(yield - (1.0_C_DOUBLE - FluorYield(82, K_SHELL))) < 1E-6_C_DOUBLE)

yield = AugerYield(82, M3_SHELL, error)
CALL assert(ABS(yield - 0.1719525_C_DOUBLE) < 1E-6_C_DOUBLE)
CALL assert(.NOT. ASSOCIATED(error))

yield = AugerYield(-35, K_SHELL, error)
CALL assert(yield == 0.0)
CALL assert(ASSOCIATED(error))
CALL assert(error%code == XRL_ERROR_INVALID_ARGUMENT)
WRITE (output_unit, '(A,A)') 'Error message: ', TRIM(error%message)
DEALLOCATE(error)

yield = AugerYield(82, N2_SHELL, error)
CALL assert(yield == 0.0)
CALL assert(ASSOCIATED(error))
CALL assert(error%code == XRL_ERROR_INVALID_ARGUMENT)
WRITE (output_unit, '(A,A)') 'Error message: ', TRIM(error%message)
DEALLOCATE(error)

yield = AugerYield(26, M5_SHELL, error)
CALL assert(yield == 0.0)
CALL assert(ASSOCIATED(error))
CALL assert(error%code == XRL_ERROR_INVALID_ARGUMENT)
WRITE (output_unit, '(A,A)') 'Error message: ', TRIM(error%message)
DEALLOCATE(error)

ENDPROGRAM test_auger

File: java/tests/TestAuger.java
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;
import org.junit.jupiter.api.Test;
import com.github.tschoonj.xraylib.Xraylib;

public class TestAuger {
	@Test
	public void test_rate_Pb_K_L3M5() {
		double rate = Xraylib.AugerRate(82, Xraylib.K_L3M5_AUGER);
		assertEquals(rate, 0.004573193387, 1E-6);
	}

	@Test
	public void test_rate_Pb_L3_M4N7() {
		double rate = Xraylib.AugerRate(82, Xraylib.L3_M4N7_AUGER);
		assertEquals(rate, 0.0024327572005, 1E-6);
	}

	@Test
	public void test_rate_bad_Z() {
		IllegalArgumentException exc = assertThrows(IllegalArgumentException.class, () -> {
			double rate = Xraylib.AugerRate(-35, Xraylib.L3_M4N7_AUGER);
		});
		assertEquals(exc.getMessage(), Xraylib.Z_OUT_OF_RANGE);
	}

	@Test
	public void test_rate_bad_trans() {
		IllegalArgumentException exc = assertThrows(IllegalArgumentException.class, () -> {
			double rate = Xraylib.AugerRate(82, Xraylib.M4_M5Q3_AUGER + 1);
		});
		assertEquals(exc.getMessage(), Xraylib.UNKNOWN_AUGER);
	}

	@Test
	public void test_rate_invalid_trans() {
		IllegalArgumentException exc = assertThrows(IllegalArgumentException.class, () -> {
			double rate = Xraylib.AugerRate(62, Xraylib.L3_M4N7_AUGER);
		});
		assertEquals(exc.getMessage(), Xraylib.INVALID_AUGER);
	}

	@Test
	public void test_yield_Pb_K() {
		double ayield = Xraylib.AugerYield(82, Xraylib.K_SHELL);
		assertEquals(ayield, 1.0 - Xraylib.FluorYield(82, Xraylib.K_SHELL));
	}

	@Test
	public void test_yield_Pb_M3() {
		double ayield = Xraylib.AugerYield(82, Xraylib.M3_SHELL);
		assertEquals(ayield, 0.1719525, 1E-6);
	}

	@Test
	public void test_yield_Pb_L1() {
		double ayield = Xraylib.AugerYield(82, Xraylib.L1_SHELL);
		assertEquals(ayield, 0.1825, 1E-6);
	}

	@Test
	public void test_yield_bad_Z() {
		IllegalArgumentException exc = assertThrows(IllegalArgumentException.class, () -> {
			double ayield = Xraylib.AugerYield(-35, Xraylib.K_SHELL);
		});
		assertEquals(exc.getMessage(), Xraylib.Z_OUT_OF_RANGE);
	}

	@Test
	public void test_yield_invalid_shell() {
		IllegalArgumentException exc = assertThrows(IllegalArgumentException.class, () -> {
			double ayield = Xraylib.AugerYield(26, Xraylib.M5_SHELL);
		});
		assertEquals(exc.getMessage(), Xraylib.INVALID_SHELL);
	}

	@Test
	public void test_yield_bad_shell() {
		IllegalArgumentException exc = assertThrows(IllegalArgumentException.class, () -> {
			double ayield = Xraylib.AugerYield(82, -5);
		});
		assertEquals(exc.getMessage(), Xraylib.UNKNOWN_SHELL);
	}
}
